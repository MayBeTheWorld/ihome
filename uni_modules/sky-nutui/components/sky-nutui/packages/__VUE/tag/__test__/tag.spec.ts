import Tag from '../index.vue';
import { mount, config } from '@vue/test-utils';

import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should emit close event when clicking the close icon', () => {
  const wrapper = mount(Tag, {
    props: {
      closeable: true
    }
  });

  wrapper.find('.nut-tag--close').trigger('click');
  expect(wrapper.emitted('close')!.length).toEqual(1);
});

test('should hide tag when the show prop is false', () => {
  const wrapper = mount(Tag, {
    props: {
      show: false
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should not trigger click event when clicking the close icon', () => {
  const onClick = jest.fn();
  const wrapper = mount(Tag, {
    props: {
      onClick,
      closeable: true
    }
  });

  wrapper.find('.nut-tag--close').trigger('click');
  expect(onClick).toHaveBeenCalledTimes(0);

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('should render textColor correctly', () => {
  const wrapper = mount(Tag, {
    props: {
      plain: true,
      color: 'red',
      textColor: 'blue'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});
