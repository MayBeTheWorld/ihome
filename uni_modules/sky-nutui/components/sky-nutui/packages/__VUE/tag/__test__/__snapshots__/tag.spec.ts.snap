// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide tag when the show prop is false 1`] = `
"<view class=\\"nut-tag nut-tag--default\\" show=\\"false\\">
  <!--v-if-->
</view>"
`;

exports[`should render textColor correctly 1`] = `
"<view class=\\"nut-tag nut-tag--default nut-tag--plain\\" style=\\"background: red; color: blue;\\">
  <!--v-if-->
</view>"
`;
